### DISCUSSION TITLE
read multi-model time-lagged ensemble (HREF) with ensemblestat

### DISCUSSION THREAD

**Discussion by JiliDong-NOAA** [dtcenter/METplus#1598] [Configuration] [answered]
Hi,

I am trying to set up the config for emsemblestat to process HREF which is a multi-model time-lagged ensemble. The ensemble will look like below for the valid time 2021/05/03/12 Z:

m1_2021050300f12.grb2
m2_2021050300f12.grb2
m3_2021050212f24.grb2
m4_2021050218f18,grb2

m1-m4 are four different models: two models have no time lag; one model has -12h and one has -6h time lag.

The setup I am trying to mimic is from METPLUS example for HRRR time-lagged ensemble HRRR-TLE:

```
LOOP_BY = VALID
VALID_TIME_FMT = %Y%m%d%H
VALID_BEG=2021050312
VALID_END=2021050312
VALID_INCREMENT=86400
LEAD_SEQ = begin_end_incr(12, 0, -6)
FCST_MTD_INPUT_TEMPLATE= {init?fmt=%Y%m%d}/{init?fmt=%Y%m%d}_i{init?fmt=%H}_f{lead?fmt=%3H}_HRRRTLE_PHPT.grb2
```
But it looks like the above setup is for a single model time-lagged ensemble and I don't know how to modify it to read in multiple models with different lags.

Thanks,

Jili

**Comment by jprestop** [comment, marked as answer]
Hi @JiliDong-NOAA.  Thank you for your question. In [this section](https://metplus.readthedocs.io/en/latest/generated/model_applications/convection_allowing_models/EnsembleStat_fcstHRRRE_obsHRRRE_Sfc_MultiField.html?highlight=FCST_ENSEMBLE_STAT_INPUT_TEMPLATE#metplus-configuration), you'll find the following setting, which shows an example of a comma-separated list of ensemble members:

```
# input ensemble template - comma separated list of ensemble members
FCST_ENSEMBLE_STAT_INPUT_TEMPLATE =
    {init?fmt=%Y%m%d%H}/postprd_mem0001/wrfprs_conus_mem0001_{lead?fmt=%HH}.grib2,
    {init?fmt=%Y%m%d%H}/postprd_mem0002/wrfprs_conus_mem0002_{lead?fmt=%HH}.grib2
```
You would add as many entries to this list as you'd need and could use "offsets" to make it time-lagged.

Here is an example of [Shifting Times in Filename Templates](https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#shifting-times-in-filename-templates) in our documentation.  

I hope that helps!  Please let us know if you're able to get it working or if you need more assistance.

**Comment by JiliDong-NOAA** [comment]
Thanks @jprestop! Combining the member list with shift/offset for individual members works well and perfectly solved my problem.

**Reply by jprestop** [reply]
That's great, @JiliDong-NOAA.  I'm so glad to hear it.  Thank you!  I will go ahead and lock this Discussion to prevent future posts. That's how we encourage users to open new discussions for new questions. Feel free to do so if more questions arise.
